qiskit/algorithms/fidelities/fidelity.py

```diff
@@ -0,0 +1,146 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Compute-uncompute fidelity interface using primitives
+"""
+
+from __future__ import annotations
+from typing import Sequence
+
+from qiskit import QuantumCircuit
+from qiskit.primitives import Sampler
+from qiskit.primitives.primitive_job import PrimitiveJob
+from .base_fidelity import BaseFidelity
+from .fidelity_result import FidelityResult
+
+class Fidelity(BaseFidelity):
+    """
+    This class leverages the sampler primitive to calculate the fidelity of two quantum circuits
+    with the compute-uncompute method.
+    """
+
+    def __init__(self, sampler: Sampler, **run_options) -> None:
```

Comment:
I admit I have not looked at the primitives for a bit. This is how things are being done to gather everything else here as some default rather than an explicit dictionary. SO these are not Fidelity run_options rather run options to pass into Sampler. If Sampler has default run options these could be supplied there too I guess, unless we are sharing some sampler instance and want different defaults to overide them

Reply:
We are in discussion.
https://github.com/Qiskit/qiskit-terra/pull/8513
Primitives will use dataclass as settings like `runtime` primitives.
https://github.com/Qiskit/qiskit-ibm-runtime/blob/a8ae6fde3998e2eeb1a79130650c97875758fc43/qiskit_ibm_runtime/options.py